### ISSUE TITLE
concurrently fails at runtime with default export

### ISSUE THREAD

**Issue by alamothe** [evanw/esbuild#2487]
Hello,
My TypeScript code is this:
```tsx
import concurrently from "concurrently"

const { result } = await concurrently(["npm install"])
await result
```

This works fine when I execute it with `ts-node` etc. but when I ran it through ESBuild it fails at runtime:
```
Type Error: rb is not a function
```

By looking at the generated code:
```js
  var concurrently_exports = {};
  __export(concurrently_exports, {
    default: () => concurrently_default
  });
  __reExport(concurrently_exports, __toModule(require_src()));
  var import_src = __toModule(require_src());
  var concurrently_default = import_src.default.default;
```
It looks like there is a double default. If I remove one default, it will work. 

Where is the problem?

**Comment by hyrious** [comment]
The double default comes from

1. The source code of concurrently includes one `.default`: https://github.com/open-cli-tools/concurrently/blob/main/index.mjs#L10

2. When esbuild wraps a cjs module (concurrently/dist/src/index.js) to esm context, it transforms esm `default` import into cjs `[cjsmodule].default`, i.e.

   ```js
   import concurrently from './dist/src/index.js';
   // -> all_exports = wrap_cjs_module(./dist/src/index.js)
   // -> concurrently = all_exports.default
   export default concurrently.default;
   // -> all_exports.default.default
   ```

Why this breaks esbuild cjs-esm-interop? Because esbuild already takes care of cjs modules and ideally people don't have to write any `.default` in their code. The code above (import a cjs module in esm context and export its .default property) only works in node.js runtime.

`ts-node` doesn't error to you because it only transforms typescript sources and the other code are still run in native node.js. So the solution is add `--external:concurrenly` to skip bundling it.

**Comment by evanw** [comment]
How do you reproduce this exactly? This happens if I try to run your example verbatim:

```
$ esbuild --bundle your-example.ts --platform=node --format=esm --outfile=out.mjs
$ node out.mjs
file://./out.mjs:12
  throw new Error('Dynamic require of "' + x + '" is not supported');
        ^

Error: Dynamic require of "assert" is not supported
```

This happens because I'm using the ESM format (due to top-level await) and ESM doesn't come with `require`. And if I work around this problem by avoiding top-level await and using CommonJS instead, this happens instead:

```
$ esbuild --bundle your-example.ts --platform=node --format=esm --outfile=out.js
$ node out.js

  out.js  776.1kb

⚡ Done in 174ms
[0] 
[0] up to date, audited 31 packages in 519ms
[0] 
[0] 5 packages are looking for funding
[0]   run `npm fund` for details
[0] 
[0] found 0 vulnerabilities
[0] npm install exited with code 0
```

Everything works fine. So there's no problem? The code was changed to this instead (no top-level await):

```ts
import concurrently from "concurrently"
const { result } = concurrently(["npm install"])
```

That makes sense because the `concurrently/index.mjs` file should trigger esbuild's node compatibility mode due to the `.mjs` extension. This is documented here: https://esbuild.github.io/content-types/#default-interop.

I can only reproduce your problem with a really old build of esbuild from last year. The last build of esbuild with this problem was 0.14.3, before it was fixed in 0.14.4. So I'm guessing you are using a very outdated version of esbuild and you should update it to fix your problem.

**Comment by alamothe** [comment]
Thank you very much for your responses. 

Just to give context, I am developing a plug-in for Yarn 3. As such I am using a package `@yarnpkg/builder` they provided. It uses `iife` format.

You are correct that they indeed use an ancient version: https://github.com/yarnpkg/berry/blob/master/packages/yarnpkg-builder/package.json:
```
    "esbuild": "npm:esbuild-wasm@^0.11.20",
```
I am not sure why this is the case.
